crates/key-value-azure/src/store.rs

```diff
@@ -216,24 +228,67 @@ impl Store for AzureCosmosStore {
         Ok(())
     }
 
+    // WARNING: this function only works on the primary region because the
+    // `azure_data_cosmos-0.21.0` release of the Azure cosmos SDK does not
+    // support setting allow_tentative_writes on patch requests. The initial
+    // value for the item must be set through this interfaces, as this sets the
+    // number value if it does not exist. If the value was previously set using
+    // the `set` interface, this will fail due to a type mismatch.
+    //
+    // TODO: The function should parse the new value from the return response
+    // rather than sending an additional new request. However, the current SDK
+    // version does not support this.
     async fn increment(&self, key: String, delta: i64) -> Result<i64, Error> {
         let operations = vec![Operation::incr("/value", delta).map_err(log_error)?];
-        let _ = self
+        match self
             .client
-            .document_client(key.clone(), &self.store_id)
+            .document_client(&key, &self.store_id.clone().unwrap_or(key.to_string()))
             .map_err(log_error)?
             .patch_document(operations)
             .await
-            .map_err(log_error)?;
-        let pair = self.get_pair(key.as_ref()).await?;
-        match pair {
-            Some(p) => Ok(i64::from_le_bytes(
-                p.value.try_into().expect("incorrect length"),
-            )),
-            None => Err(Error::Other(
-                "increment returned an empty value after patching, which indicates a bug"
-                    .to_string(),
-            )),
+        {
+            Err(e) => {
+                if e.as_http_error()
+                    .map(|e| e.status())
```

Comment:
It might help with @itowlson's comment directly below this to do something like:

```suggestion
            Err(e)  if e.as_http_error().map(|e| e.status() == 404).umwrap_or(false) => {
               
```